examples/diff.c

```diff
@@ -335,3 +349,58 @@ static void diff_print_stats(git_diff *diff, struct opts *o)
 	git_buf_free(&b);
 	git_diff_stats_free(stats);
 }
+
+int string_to_int(const char * cmd_opt)
+{
+	int to_return = 0;
+	int i = 0;
+
+
+	while(cmd_opt[i]!='%') {
+		to_return *= 10;
+		to_return += cmd_opt[i] - '0';
+		i++;
+	}
+	return to_return;
+}
```

Comment:
Do whatever git does. It may error out, let it through or reset to some default. Look at the code of this command in git to see how it deals with its arguments.